packages/appstore/src/components/currency-switcher-card/demo/demo-account-card.tsx

```diff
@@ -6,16 +6,34 @@ import BalanceText from 'Components/elements/text/balance-text';
 import { useStores } from 'Stores/index';
 import './demo-account-card.scss';
 import { localize } from '@deriv/translations';
+import {
+    useCFDDemoAccounts,
+    useCurrencyExchangeRate,
+    usePlatformDemoAccount,
+    useTotalAccountBalance,
+} from '@deriv/hooks';
 
 const DemoAccountCard = () => {
     const { client, traders_hub } = useStores();
-    const { accounts, loginid, resetVirtualBalance } = client;
-    const { platform_demo_balance, selected_account_type } = traders_hub;
+    const { accounts, loginid, resetVirtualBalance, default_currency } = client;
+    const { selected_account_type } = traders_hub;
 
     const canResetBalance = () => {
         return accounts[loginid]?.balance !== 10000;
     };
 
+    const platform_demo_account = usePlatformDemoAccount();
+    const cfd_demo_accounts = useCFDDemoAccounts();
+    const cfd_demo_balance = useTotalAccountBalance(cfd_demo_accounts);
+    const cfd_demo_rate = useCurrencyExchangeRate(platform_demo_account.currency);
+
+    const DemoAccounttotalBalance = React.useMemo(() => {
+        return {
+            balance: platform_demo_account.balance + cfd_demo_balance.balance * cfd_demo_rate,
+            currency: platform_demo_account.currency,
+        };
+    }, [cfd_demo_balance.balance, cfd_demo_rate, platform_demo_account.balance, platform_demo_account.currency]);
```

Comment:
```suggestion
    const DemoAccountTotalBalance = React.useMemo(
        () => (platform_demo_account?.balance || 0) + cfd_demo_balance.balance * cfd_demo_rate,
        [cfd_demo_balance.balance, cfd_demo_rate, platform_demo_account]
    );
```
@shayan-deriv But TS complains that `balance` doesn't exist on`platform_demo_account` 🤔 Not sure if it's a dead code or the type is not correct 🤔

Reply:
instead of `platform_demo_account` I should put `platform_demo_account?.balance` in the dependency list. otherwise It doesn't re-render after resetting demo balance